src/main/java/org/jabref/gui/preferences/websearch/WebSearchTabViewModel.java

```diff
@@ -181,57 +181,63 @@ public BooleanProperty getApikeyPersistProperty() {
-                int statusCode = ((HttpURLConnection) urlDownload.getSource().openConnection()).getResponseCode();
-                keyValid = (statusCode >= 200) && (statusCode < 300);
+            final String testUrlWithoutApiKey = fetcherOpt.get().getTestUrl();
+            if (testUrlWithoutApiKey == null) {
+                dialogService.showWarningDialogAndWait(
+                        Localization.lang("Check %0 API Key Setting", apiKeyName),
+                        Localization.lang("Fetcher cannot be tested!"));
+                return;
+            }
 
-                URLDownload.setSSLVerification(defaultSslSocketFactory, defaultHostnameVerifier);
-            } catch (IOException | UnirestException e) {
+            final String apiKey = selectedApiKeyProperty.get().getKey();
+
+            boolean keyValid;
+            if (!apiKey.isEmpty()) {
+                URLDownload urlDownload;
+                try {
+                    SSLSocketFactory defaultSslSocketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
+                    HostnameVerifier defaultHostnameVerifier = HttpsURLConnection.getDefaultHostnameVerifier();
+
+                    urlDownload = new URLDownload(testUrlWithoutApiKey + apiKey);
+                    // The HEAD request cannot be used because its response is not 200 (maybe 404 or 596...).
+                    int statusCode = ((HttpURLConnection) urlDownload.getSource().openConnection()).getResponseCode();
+                    keyValid = (statusCode >= 200) && (statusCode < 300);
+
+                    URLDownload.setSSLVerification(defaultSslSocketFactory, defaultHostnameVerifier);
+                } catch (
+                        IOException |
+                        UnirestException e) {
+                    keyValid = false;
+                }
+            } else {
                 keyValid = false;
             }
-        } else {
-            keyValid = false;
-        }
 
-        if (keyValid) {
-            dialogService.showInformationDialogAndWait(Localization.lang("Check %0 API Key Setting", apiKeyName), Localization.lang("Connection successful!"));
-        } else {
-            dialogService.showErrorDialogAndWait(Localization.lang("Check %0 API Key Setting", apiKeyName), Localization.lang("Connection failed!"));
+            if (keyValid) {
+                dialogService.showInformationDialogAndWait(Localization.lang("Check %0 API Key Setting", apiKeyName), Localization.lang("Connection successful!"));
+            } else {
+                dialogService.showErrorDialogAndWait(Localization.lang("Check %0 API Key Setting", apiKeyName), Localization.lang("Connection failed!"));
+            }
+        } catch (Exception e) {
+                dialogService.showErrorDialogAndWait(Localization.lang("No API was selected"), Localization.lang("Please select an API first"));
         }
```

Comment:
Never use try-catch for program logic. And keep the scope of the try block always as small as possible, also the Exceptions as percise as possible. See https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#9-exceptions